resources/js/components/shared/TimelinePlot.vue

```diff
@@ -0,0 +1,393 @@
+<template>
+  <div id="timeline_plot"></div>
+</template>
+<script>
+export default {
+  name: "TimelinePlot",
+  props: {
+    plotData: {
+      type: Object,
+      required: true,
+      default: () => { return null }
+    },
```

Comment:
You could consider adding a [validator](https://vuejs.org/guide/components/props.html#prop-validation) to verify that the input data is valid.

Reply:
From the doc page you provided, it seems that this will only apply for dev mode:
> When prop validation fails, Vue will produce a console warning (if using the development build).

Is it still worth adding?

Reply:
Good catch.  I don't think it's worth adding.  The long-term solution to this problem is to simply use TypeScript, but I think what you have is good enough for now.